standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -2107,15 +2115,17 @@ private ColumnStatisticsObj prepareCSObjWithAdjustedNDV(Object[] row, int i,
                 makeParams(inputColNames.size()), makeParams(inputPartNames.size()));
             long start = doTrace ? System.nanoTime() : 0;
             Query query = pm.newQuery("javax.jdo.query.SQL", queryText);
-            Object qResult = executeWithArray(query, prepareParams(
-                catName, dbName, tableName, inputPartNames, inputColNames, engine), queryText);
-            MetastoreDirectSqlUtils.timingTrace(doTrace, queryText0, start, (doTrace ? System.nanoTime() : 0));
-            if (qResult == null) {
-              query.closeAll();
-              return Collections.emptyList();
+            try {
```

Comment:
I think `addQueryAfterUse` is there for closing the queries. So we might not need them anymore. What do you think?

Reply:
Maybe we should need this,  `Query#execute` may return a lazy result, e.g, the `org.datanucleus.store.rdbms.query.ForwardQueryResult`,  closing the query ealier may result to the exception `rs is closed` when iterating/accessing the result.

Reply:
Sounds like a bad practice, but we do not want to solve every problem here 😄